src/meta-srv/src/region/supervisor.rs

```diff
@@ -389,75 +426,131 @@ impl RegionSupervisor {
             .context(error::MaintenanceModeManagerSnafu)
     }
 
-    async fn do_failover(&mut self, datanode_id: DatanodeId, region_id: RegionId) -> Result<()> {
-        let count = *self
-            .failover_counts
-            .entry((datanode_id, region_id))
-            .and_modify(|count| *count += 1)
-            .or_insert(1);
+    async fn select_peers(
+        &self,
+        from_peer_id: DatanodeId,
+        regions: &[RegionId],
+        failure_datanodes: &[DatanodeId],
+    ) -> Result<Vec<(RegionId, Peer)>> {
+        let exclude_peer_ids = HashSet::from_iter(failure_datanodes.iter().cloned());
+        match &self.selector {
+            RegionSupervisorSelector::NaiveSelector(selector) => {
+                let opt = SelectorOptions {
+                    min_required_items: regions.len(),
+                    allow_duplication: true,
+                    exclude_peer_ids,
+                };
+                let peers = selector.select(&self.selector_context, opt).await?;
+                ensure!(
+                    peers.len() == regions.len(),
+                    error::PeerCountMismatchSnafu {
```

Comment:
error:NoEnoughAvailableNode

Reply:
Resolved in [6f6b0c8](https://github.com/GreptimeTeam/greptimedb/pull/5990/commits/6f6b0c8cf0af29944c97c77c19d20822131aa0eb)